### PR TITLE
libvirt: 7.0.0 -> 7.7.0

### PR DESCRIPTION
**Pull request by Kranzes**
###### Motivation for this change
Version update that should have been done a while ago.

- https://libvirt.org/news.html#v7-7-0-2021-09-01
- https://gitlab.com/libvirt/libvirt-python/-/compare/v7.0.0...v7.7.0
- https://gitlab.com/libvirt/libvirt-perl/-/compare/v7.0.0...v7.7.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130805` at 638b7232 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>libvmi</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/df4f5379652d51948498bed6f8fd0084">collectd</a></li>
    <li><a href="https://gist.github.com/8091e0acde84a3ad94c61a4b407e90a8">collectd-data</a></li>
    <li><a href="https://gist.github.com/25e4f69b1d6def1d0e01999536b6751e">docker-machine-kvm</a></li>
    <li><a href="https://gist.github.com/22ca9b3cf9ae9093f06d99faaf63ede5">docker-machine-kvm2</a></li>
    <li><a href="https://gist.github.com/6d523d187ef1a0a7a412a1bf2228d0f9">gnome.gnome-boxes</a></li>
    <li><a href="https://gist.github.com/23ec6ec28ba6fd49ea1bba47b31d2ecf">haskellPackages.libvirt-hs</a></li>
    <li><a href="https://gist.github.com/63a832eae13cef91b4fb5fad9f22784c">libguestfs</a></li>
    <li><a href="https://gist.github.com/d896e4e2dcf2b27f7551d37ba2c2ba91">libvirt</a></li>
    <li><a href="https://gist.github.com/eb7ed8205d94b4b587ee624536446d83">libvirt-glib</a></li>
    <li><a href="https://gist.github.com/183a1d9c25042d770963c0aa997f403e">minikube</a></li>
    <li><a href="https://gist.github.com/a88ec722fe9771a99600f14632e92fd2">minishift</a></li>
    <li><a href="https://gist.github.com/3e6fca0452819e08a73849732395407c">nixopsUnstable</a></li>
    <li><a href="https://gist.github.com/afa8bee47f40ae55918ff1c2e646d14e">perl530Packages.SysVirt</a></li>
    <li><a href="https://gist.github.com/968cb84e71fd2ce73ee30c0c4852bf68">perl532Packages.SysVirt</a></li>
    <li><a href="https://gist.github.com/884bb4c97d476e268d63fbda1d925de0">python38Packages.guestfs</a></li>
    <li><a href="https://gist.github.com/b6a6a7e5f21dc282062452d159b7b1e4">python38Packages.libvirt</a></li>
    <li><a href="https://gist.github.com/db016989bb8918a9529205b727cb8c88">python39Packages.guestfs</a></li>
    <li><a href="https://gist.github.com/93d279cefe7c885e57b7c14d50e651fe">python39Packages.libvirt</a></li>
    <li><a href="https://gist.github.com/c2143e83e2a9849c90a324e179f03fc8">rubyPackages.ruby-libvirt</a> (rubyPackages_2_7.ruby-libvirt)</li>
    <li><a href="https://gist.github.com/09221d502ecda6c7162c82635ad7377c">rubyPackages_2_6.ruby-libvirt</a></li>
    <li><a href="https://gist.github.com/4e362ba28f6f499d78d42a0d78ffd6f9">rubyPackages_3_0.ruby-libvirt</a></li>
    <li><a href="https://gist.github.com/0360b6589a71d5f22d33bcdb3ba016d3">terraform-full</a></li>
    <li><a href="https://gist.github.com/412073ac0a020d16d444f3c1f9fae924">terraform-providers.libvirt</a></li>
    <li><a href="https://gist.github.com/60b6947048daa85f9e700346e13f49b9">vagrant</a></li>
    <li><a href="https://gist.github.com/773c87b767f0cf24cfdd8af84b15b26e">virt-manager</a></li>
    <li><a href="https://gist.github.com/377b8ac18ff56907c1d3224834f010b4">virt-manager-qt</a></li>
    <li><a href="https://gist.github.com/4fa8782af6f402a3c1221da6edf4b131">virt-top</a></li>
    <li><a href="https://gist.github.com/3be0299287bf1ba73cbaaa618b7f2b6e">virt-viewer</a></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`lgpl2` is a deprecated license, please check if project uses `lgpl2Plus` or `lgpl2Only` and change `meta.license` accordingly.

Near pkgs/development/python-modules/libvirt/default.nix:25:5:
```
   |
25 |     license = licenses.lgpl2;
   |     ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/libvirt/default.nix [hunk 1]
```diff
@@ -32,19 +32,19 @@ let
...
 
   src =
     if buildFromTarball then
       fetchurl {
         url = "https://libvirt.org/sources/${pname}-${version}.tar.xz";
-        sha256 = "12fxkpy7j2qhfxypw9jg3bzdd9xx6vf6x96iy5kjihh89n236f6a";
+        sha256 = "sha256-uKEu9zSg6HrHx7yb8H25Yvt8r6yAGcXVTGNCZc+CDNY=";
       }
     else
       fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libvirt/default.nix:44]: Can we use fetchFromGitLab here?

**Comment by risicle** [comment]
Fails to build for me, nixos x86_64:

```
builder for '/nix/store/scwrpw4favc5012jmcxsm1iq8b2x1jxb-libvirt-7.5.0.drv' failed with exit code 1; last 10 log lines:
    File "/nix/store/6cfajs6lsy9b4wxp3jvyyl1g5x2pjmpr-python3-3.8.9/lib/python3.8/os.py", line 213, in makedirs
      makedirs(head, exist_ok=exist_ok)
    File "/nix/store/6cfajs6lsy9b4wxp3jvyyl1g5x2pjmpr-python3-3.8.9/lib/python3.8/os.py", line 223, in makedirs
      mkdir(name, mode)
  PermissionError: [Errno 13] Permission denied: '/var'

  FAILED: install script '/build/libvirt/scripts/meson-python.sh /nix/store/6cfajs6lsy9b4wxp3jvyyl1g5x2pjmpr-python3-3.8.9/bin/python3 /build/libvirt/scripts/meson-install-dirs.py /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/log/libvirt /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/lockd /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/lockd/files /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/run/libvirt/lockd /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/network /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/dnsmasq /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/run/libvirt/network /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/lxc /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/run/libvirt/lxc /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/log/libvirt/lxc /var/lib/libvirt/ch /run/libvirt/ch /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/qemu /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/run/libvirt/qemu /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/cache/libvirt/qemu /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/log/libvirt/qemu /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/swtpm /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/run/libvirt/qemu/swtpm /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/log/swtpm/libvirt/qemu /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/cache/libvirt /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/images /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/filesystems /nix/store/hkbxlwlpb7scl87fwalynx9cwxicrnwa-libvirt-7.5.0/var/lib/libvirt/boot' exit code 1, stopped
  FAILED: meson-install
  /nix/store/2ivahgplffg59pgkm1vfkiwp91gqhg3r-meson-0.57.1/bin/meson install --no-rebuild
  ninja: build stopped: subcommand failed.
```

File: pkgs/development/libraries/libvirt/default.nix (hunk 1 above)
Comment (Kranzes) [reply pkgs/development/libraries/libvirt/default.nix:44]: Wanted to, but it seems like fetchFromGitLab does not have fetchSubmodules...

**Comment by Kranzes** [comment]
@risicle I am very well aware that it is failing to build, I have to change a couple of things in the patch file but i am not sure what i need to change.

File: pkgs/development/libraries/libvirt/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/libvirt/default.nix:44]: I added those options in #131047

**Comment by Kranzes** [comment]
Is anyone able to help with this?

**Comment by anton-dessiatov** [comment]
@Kranzes, the following patch for your MR seems to do the trick for me:

<details>
<summary> Patch </summary>

```
diff --git a/pkgs/development/libraries/libvirt/0001-meson-patch-in-an-install-prefix-for-building-on-nix.patch b/pkgs/development/libraries/libvirt/0001-meson-patch-in-an-install-prefix-for-building-on-nix.patch
index 86dd461ea07..bd00c66f85a 100644
--- a/pkgs/development/libraries/libvirt/0001-meson-patch-in-an-install-prefix-for-building-on-nix.patch
+++ b/pkgs/development/libraries/libvirt/0001-meson-patch-in-an-install-prefix-for-building-on-nix.patch
@@ -1,27 +1,5 @@
-From a896b0be849455edb83a9305dfec9b41447ef3e4 Mon Sep 17 00:00:00 2001
-From: Euan Kemp <euank@euank.com>
-Date: Thu, 14 Jan 2021 00:32:00 -0800
-Subject: [PATCH] meson: patch in an install prefix for building on nix
-
-Used in the nixpkgs version of libvirt so that we can install things in
-the nix store, but read them from the root filesystem.
----
- meson.build                       |  9 +++++++++
- meson_options.txt                 |  2 ++
- src/libxl/meson.build             |  6 +++---
- src/locking/meson.build           |  8 ++++----
- src/lxc/meson.build               |  6 +++---
- src/meson.build                   | 18 +++++++++---------
- src/network/meson.build           | 12 ++++++------
- src/nwfilter/xml/meson.build      |  2 +-
- src/qemu/meson.build              | 14 +++++++-------
- src/remote/meson.build            |  6 +++---
- src/security/apparmor/meson.build |  8 ++++----
- tools/meson.build                 |  4 ++--
- 12 files changed, 53 insertions(+), 42 deletions(-)
-
 diff --git a/meson.build b/meson.build
-index b5164f6..33719f1 100644
+index 1fab424fde..7f1d23e03e 100644
 --- a/meson.build
 +++ b/meson.build
 @@ -39,6 +39,8 @@ if host_machine.system() == 'windows'
@@ -48,7 +26,7 @@ index b5164f6..33719f1 100644
  # sysconfdir as this makes a lot of things break in testing situations
  if prefix == '/usr'
 diff --git a/meson_options.txt b/meson_options.txt
-index e5d79c2..081cd32 100644
+index 859ed36b8f..c8aa8fd72a 100644
 --- a/meson_options.txt
 +++ b/meson_options.txt
 @@ -1,3 +1,5 @@
@@ -57,8 +35,22 @@ index e5d79c2..081cd32 100644
  option('no_git', type: 'boolean', value: false, description: 'Disable git submodule update')
  option('packager', type: 'string', value: '', description: 'Extra packager name')
  option('packager_version', type: 'string', value: '', description: 'Extra packager version')
+diff --git a/src/ch/meson.build b/src/ch/meson.build
+index e34974d56c..4767763c2c 100644
+--- a/src/ch/meson.build
++++ b/src/ch/meson.build
+@@ -68,7 +68,7 @@ if conf.has('WITH_CH')
+   }
+ 
+   virt_install_dirs += [
+-    localstatedir / 'lib' / 'libvirt' / 'ch',
+-    runstatedir / 'libvirt' / 'ch',
++    install_prefix + localstatedir / 'lib' / 'libvirt' / 'ch',
++    install_prefix + runstatedir / 'libvirt' / 'ch',
+   ]
+ endif
 diff --git a/src/libxl/meson.build b/src/libxl/meson.build
-index 3bb6cc5..78d7be0 100644
+index 9793899106..4904387162 100644
 --- a/src/libxl/meson.build
 +++ b/src/libxl/meson.build
 @@ -84,8 +84,8 @@ if conf.has('WITH_LIBXL')
@@ -74,7 +66,7 @@ index 3bb6cc5..78d7be0 100644
    ]
  endif
 diff --git a/src/locking/meson.build b/src/locking/meson.build
-index 8a28310..9da81cc 100644
+index 184d3c3f56..7ffd3740e3 100644
 --- a/src/locking/meson.build
 +++ b/src/locking/meson.build
 @@ -243,14 +243,14 @@ if conf.has('WITH_LIBVIRTD')
@@ -97,10 +89,10 @@ index 8a28310..9da81cc 100644
    endif
  endif
 diff --git a/src/lxc/meson.build b/src/lxc/meson.build
-index f8e2a88..96d6687 100644
+index ad5c659dba..8010dfbf02 100644
 --- a/src/lxc/meson.build
 +++ b/src/lxc/meson.build
-@@ -182,8 +182,8 @@ if conf.has('WITH_LXC')
+@@ -181,8 +181,8 @@ if conf.has('WITH_LXC')
    }
  
    virt_install_dirs += [
@@ -113,7 +105,7 @@ index f8e2a88..96d6687 100644
    ]
  endif
 diff --git a/src/meson.build b/src/meson.build
-index 7c47821..d33d16a 100644
+index 2bd88e6699..76df507968 100644
 --- a/src/meson.build
 +++ b/src/meson.build
 @@ -669,7 +669,7 @@ endforeach
@@ -176,10 +168,10 @@ index 7c47821..d33d16a 100644
  
  meson.add_install_script(
 diff --git a/src/network/meson.build b/src/network/meson.build
-index 3ec598c..b02040b 100644
+index d6fb624bb7..96b757b144 100644
 --- a/src/network/meson.build
 +++ b/src/network/meson.build
-@@ -79,9 +79,9 @@ if conf.has('WITH_NETWORK')
+@@ -78,9 +78,9 @@ if conf.has('WITH_NETWORK')
    }
  
    virt_install_dirs += [
@@ -192,7 +184,7 @@ index 3ec598c..b02040b 100644
    ]
  
    configure_file(
-@@ -89,12 +89,12 @@ if conf.has('WITH_NETWORK')
+@@ -88,12 +88,12 @@ if conf.has('WITH_NETWORK')
      output: '@BASENAME@',
      copy: true,
      install: true,
@@ -208,7 +200,7 @@ index 3ec598c..b02040b 100644
    )
  
 diff --git a/src/nwfilter/xml/meson.build b/src/nwfilter/xml/meson.build
-index 0d96c54..66c92a1 100644
+index 0d96c54ebe..66c92a1016 100644
 --- a/src/nwfilter/xml/meson.build
 +++ b/src/nwfilter/xml/meson.build
 @@ -25,4 +25,4 @@ nwfilter_xml_files = [
@@ -218,7 +210,7 @@ index 0d96c54..66c92a1 100644
 -install_data(nwfilter_xml_files, install_dir: sysconfdir / 'libvirt' / 'nwfilter')
 +install_data(nwfilter_xml_files, install_dir: install_prefix + sysconfdir / 'libvirt' / 'nwfilter')
 diff --git a/src/qemu/meson.build b/src/qemu/meson.build
-index 90640b0..8802cec 100644
+index 3898d23877..c2a2675187 100644
 --- a/src/qemu/meson.build
 +++ b/src/qemu/meson.build
 @@ -171,12 +171,12 @@ if conf.has('WITH_QEMU')
@@ -242,7 +234,7 @@ index 90640b0..8802cec 100644
    ]
  endif
 diff --git a/src/remote/meson.build b/src/remote/meson.build
-index 9ad2f6a..429a15b 100644
+index 0a188268b5..702822612e 100644
 --- a/src/remote/meson.build
 +++ b/src/remote/meson.build
 @@ -245,7 +245,7 @@ if conf.has('WITH_REMOTE')
@@ -273,7 +265,7 @@ index 9ad2f6a..429a15b 100644
    )
  endif
 diff --git a/src/security/apparmor/meson.build b/src/security/apparmor/meson.build
-index af43780..e2d6c81 100644
+index af43780211..e2d6c812f8 100644
 --- a/src/security/apparmor/meson.build
 +++ b/src/security/apparmor/meson.build
 @@ -17,22 +17,22 @@ foreach name : apparmor_gen_profiles
@@ -304,7 +296,7 @@ index af43780..e2d6c81 100644
    rename: 'usr.lib.libvirt.virt-aa-helper',
  )
 diff --git a/tools/meson.build b/tools/meson.build
-index b8c6802..dacd0ff 100644
+index 2acf7b0aaf..e1693ff200 100644
 --- a/tools/meson.build
 +++ b/tools/meson.build
 @@ -115,7 +115,7 @@ if conf.has('WITH_LOGIN_SHELL')
@@ -316,7 +308,7 @@ index b8c6802..dacd0ff 100644
  endif
  
  if host_machine.system() == 'windows'
-@@ -274,7 +274,7 @@ configure_file(
+@@ -275,7 +275,7 @@ configure_file(
  if init_script == 'systemd'
    install_data(
      'libvirt-guests.sysconf',
@@ -324,3 +316,4 @@ index b8c6802..dacd0ff 100644
 +    install_dir: install_prefix + sysconfdir / 'sysconfig',
      rename: 'libvirt-guests',
    )
+ 
```
</details>

The only truly necessary part is this:

```
+diff --git a/src/ch/meson.build b/src/ch/meson.build
+index e34974d56c..4767763c2c 100644
+--- a/src/ch/meson.build
++++ b/src/ch/meson.build
+@@ -68,7 +68,7 @@ if conf.has('WITH_CH')
+   }
+ 
+   virt_install_dirs += [
+-    localstatedir / 'lib' / 'libvirt' / 'ch',
+-    runstatedir / 'libvirt' / 'ch',
++    install_prefix + localstatedir / 'lib' / 'libvirt' / 'ch',
++    install_prefix + runstatedir / 'libvirt' / 'ch',
+   ]
+ endif
```

**Comment by Kranzes** [comment]
Could you please make a GitHub suggestion so i can commit those changes easier?

**Comment by anton-dessiatov** [comment]
I'm not sure how to make a GitHub suggestion in this case, but please take a look at MR for your fork that I've created at Kranzes/nixpkgs#1. Hope it helps.

**Comment by Kranzes** [comment]
I have almost everything ready, Including using fetchFromGitLab instead of fetchgit, and updating everything to 7.6 which came out yesterday, at the moment i am waiting for SysVirt to update to 7.6.

File: pkgs/top-level/perl-packages.nix [hunk 2]
```diff
@@ -19684,10 +19684,10 @@ let
 
   SysVirt = buildPerlModule rec {
     pname = "Sys-Virt";
-    version = "7.0.0";
+    version = "7.5.0";
```
Comment (anton-dessiatov) [inline pkgs/top-level/perl-packages.nix:19687]: Does it need to be bumped to 7.6.0?

Comment (Kranzes) [reply pkgs/top-level/perl-packages.nix:19687]: I don't know if it is *needed* but i know its *wanted*.

**Comment by jtojnar** [comment]
Would be nice to rebase to get rid of the merge commits. Otherwise looks good.

File: pkgs/top-level/perl-packages.nix (hunk 2 above)
Comment (Kranzes) [reply pkgs/top-level/perl-packages.nix:19787]: It does not seem like SysVirt will be updated to 7.6.0 any time soon, it has been about 2 weeks and I have seen no sign of it 7.6.0 being released, so I decided to stick to 7.5.0.

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 130805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>nixopsUnstable</li>
    <li>virtmanager (virt-manager)</li>
  </ul>
</details>
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>diffoscope</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>gnome.gnome-boxes</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvmi</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>perl534Packages.SysVirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_2_6.ruby-libvirt</li>
    <li>rubyPackages_3_0.ruby-libvirt</li>
    <li>terraform-full</li>
    <li>terraform-provider-libvirt (terraform-providers.libvirt)</li>
    <li>vagrant</li>
    <li>virtmanager-qt (virt-manager-qt)</li>
    <li>virt-top</li>
    <li>virtviewer (virt-viewer)</li>
    <li>virtlyst</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
Before merging we (i?) need to figure out virt-manager's build failure since that is something most people that use libvirt use.

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 130805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>nixopsUnstable</li>
  </ul>
</details>
<details>
  <summary>30 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>diffoscope</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>gnome.gnome-boxes</li>
    <li>haskellPackages.libvirt-hs</li>
    <li>libguestfs</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvmi</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>perl532Packages.SysVirt</li>
    <li>perl534Packages.SysVirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>rubyPackages.ruby-libvirt (rubyPackages_2_7.ruby-libvirt)</li>
    <li>rubyPackages_2_6.ruby-libvirt</li>
    <li>rubyPackages_3_0.ruby-libvirt</li>
    <li>terraform-full</li>
    <li>terraform-providers.libvirt</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-top</li>
    <li>virt-viewer</li>
    <li>virtlyst</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
@adisbladis @AmineChikhaoui @edolstra @rbvermaa @domenkozar, I have @'ed you because you are the maintainers of `nixopsUnstable` which is the only package that is now failing to build due to this PR's changes. Could you contribute to this PR by figuring out this build failure since it is beyond my knowledge? Thanks.

**Comment by Luflosi** [comment]
What still needs to be done to get this merged?
I had a problem with libvirt 7.0.0, where it would pass deprecated command line options to qemu, leading to an error when starting a VM. Updating libvirt fixed that.

**Comment by Kranzes** [comment]
> What still needs to be done to get this merged?
> I had a problem with libvirt 7.0.0, where it would pass deprecated command line options to qemu, leading to an error when starting a VM. Updating libvirt fixed that.

The PR of @max-privatevoid's needs to be merged first, then we will be able to build `nixopsUnstable`.

File: pkgs/top-level/perl-packages.nix (hunk 2 above)
Comment (Artturin) [reply pkgs/top-level/perl-packages.nix:19787]: Sys-Virt is now at 7.7.0

**Comment by Kranzes** [comment]
Result of `nixpkgs-review pr 130805` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>32 packages built:</summary>
  <ul>
    <li>collectd</li>
    <li>collectd-data</li>
    <li>diffoscope</li>
    <li>docker-machine-kvm</li>
    <li>docker-machine-kvm2</li>
    <li>gnome.gnome-boxes</li>
    <li>libguestfs</li>
    <li>libguestfs-with-appliance</li>
    <li>libvirt</li>
    <li>libvirt-glib</li>
    <li>libvmi</li>
    <li>minikube</li>
    <li>minishift</li>
    <li>nixopsUnstable</li>
    <li>ocamlPackages.ocaml_libvirt</li>
    <li>perl532Packages.SysVirt</li>
    <li>perl534Packages.SysVirt</li>
    <li>python38Packages.guestfs</li>
    <li>python38Packages.libvirt</li>
    <li>python39Packages.guestfs</li>
    <li>python39Packages.libvirt</li>
    <li>rubyPackages.ruby-libvirt</li>
    <li>rubyPackages_2_6.ruby-libvirt</li>
    <li>rubyPackages_3_0.ruby-libvirt</li>
    <li>terraform-full</li>
    <li>terraform-provider-libvirt (terraform-providers.libvirt)</li>
    <li>vagrant</li>
    <li>virtmanager (virt-manager)</li>
    <li>virtmanager-qt (virt-manager-qt)</li>
    <li>virt-top</li>
    <li>virtviewer (virt-viewer)</li>
    <li>virtlyst</li>
  </ul>
</details>

**Comment by Kranzes** [comment]
@ofborg eval

**Comment by Kranzes** [comment]
Should this be backported to 21.05?

**Comment by risicle** [comment]
I think that would be a pretty extreme jump given all the work this has taken.

According to the release notes the outstanding vulnerabilities involved are CVE-2021-3667 & CVE-2021-3631. The resolving commits for these are https://gitlab.com/libvirt/libvirt/-/commit/447f69dec47e1b0bd15ecd7cd49a9fd3b050fb87 and https://gitlab.com/libvirt/libvirt/-/commit/15073504dbb624d3f6c911e85557019d3620fdb2 respectively. At least both are very small (but I haven't looked into either in any depth).

**Comment by yu-re-ka** [comment]
Running on my VM host without issues 👍🏻
It also fixed some deprecated arguments to qemu, which prevented me from starting some VMs with libvirt 7.0.0.
I think it should go into master soon, you can still discuss backporting afterwards.

**Review by yu-re-ka — APPROVED** [review]

**Comment by anton-dessiatov** [comment]
I also confirm that I tested it on my Linux machine and it seems to work just fine. @Kranzes, thanks for your effort!

**Review by anton-dessiatov — APPROVED** [review]

**Comment by Kranzes** [comment]
@SuperSandro2000 can you merge?

**Pull request merged**